### DISCUSSION TITLE
EFR32MG21 and CC2652P usage

### DISCUSSION THREAD

**Discussion by lacrimal** [Koenkk/zigbee2mqtt#14261] [Devices]
Hi, 

Have EFR32MG21 and CC2652P (both Sonoff), trying to use them with not-edge version of Z2M. While CC2652P work simply solid, it has smaller transmit power. From other hand EFR32MG21 works perfectly with ZHA but in Z2M it drops devices and have problems with pairing - for example gl-c-002p has serious problems with pairing and mode change. Sometimes I need to do weird stuff with dongle and Z2M to force it to pair with EFR.
I know that EFR support in Z2M is completely experimental and that is cause of my question.

Considering stability, should I return EFR and wait for better support using stable CC2662P or switch to Z2M edge with EFR and return CC?

**Comment by jerrm** [comment]
Right now, keep the 2652 stick.   You can change the tx power to the same +20db as the "E" - https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html

But... that link is a little out of date.  I think the current default is +9db. 

I have doubts +20db offers much real world improvement.

**Comment by lacrimal** [comment]
Now something surprising - after moving to Z2M latest-dev - all problems are gone with EFR32MG21, pairing, re-pairing, switching disconnecting devices and connecting them back, and ... gl-c-002p mode change - everything works just fine.
Now I'm confused.

**Reply by yotsimitsu** [reply]
Hi, just a quick question, is the ‎GL-C-002P fully supported on Z2M? I wasn't able to find it in supported device and this thread is the only mention of it in here.

**Reply by lacrimal** [reply]
GL-C-002P is just combination of different devices as you cycle 'modes' , so it can represent itself as 006p or 009p.
More, Z2M reacts on mode change 'on the fly'. It was mentioned at least 3 times here on Z2M GitHub.

**Reply by yotsimitsu** [reply]
Ok, that makes sense and explains why I didn't find it in supported devices.
When I searched Github this was the only mentioned of this device that showed up.
Thank you

**Comment by dash042** [comment]
so should we get the EFR32MG21 or the CC2652P?

**Reply by lacrimal** [reply]
I think you should go with first one .
Main advantage is range and speed, you can't beat this with any implemenation of CC2652P.
Main disadvantage of EFR are losed connections with devices connected to bridges (it is described somewhere here). However this problem probably can be resolved with firmware upgrade.
This is from my usage perspective.

**Reply by jerrm** [reply]
I'd argue the opposite and say get a CC2652P.  

At this moment, for zigbee2mqtt, the CC2652P is objectively the better supported stick.  EZSP support is catching up, but life in z2m is easier with the CC2652P. 

Range should be about the same, both support +20db.  Some point to the lqi numbers being "better" with EFR32MG21, but znp and ezsp calculate the values/ranges differently and can't be directly compared.   Anecdotally, I can say pairing is MUCH easier with the CC26522P for me under both ZHA and Z2M.  Once paired, they both seem similar.

Looking at just the CPU/RAM specs, speed should be better on the EFR32MG21. In practice I wonder how much difference it really makes for all but the largest/busiest nets.  I can't perceive any difference between my EFR32MG21 net with 7 devices and my CC2652P net with 80+ devices.

Potentially dual protocol can be supported in the future with the EFR32MG21, I haven't seen anything about dual protocol for the CC2652P.  Personally, at $20-$30 for a second stick, I doubt I would risk the headache when/if I have the need for Thread.

**Reply by lacrimal** [reply]
So, we are not so helpful :-)

**Reply by jerrm** [reply]
I guess my best summary would be CC2652P is probably best **_TODAY_**.  ER32MG21 might be better long term.

Hopefully migration between the two will be made more user friendly and it won't be a big deal to move between sticks.

**Reply by Koenkk** [reply]
CC2652P = recommended, EFR32MG21 = experimental. So get a CC2652P .

**Reply by seblu** [reply]
I bought an EFR32MG21 few weeks ago and I have a lot of problem with it. Sometimes devices disappear from database (like #15299), sometimes the docker process goes rogue and use all the memory leading to OS crash.
It worked fine until I added devices that were sending unexpected frames.

So, I ordered CC2652P.

**Reply by freefos** [reply]
What is the status of the EFR32MG21 or is the CC2652P still the right choice?
Currently deciding between SLZB-06 and SLZB-06M.

**Reply by jerrm** [reply]
Still no z2m native backup/restore for EFR32.  Based on that alone I'd still go with CC22652P.

**Comment by Hedda** [comment]
> Hopefully migration between the two will be made more user friendly and it won't be a big deal to move between sticks.

+1 hoping that [ezsp adapter](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/ezsp) and [deconz adapte](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/deconz)r in zigbee-herdsman will get both backup and restore feature, same as the [z-stack adapter](https://github.com/Koenkk/zigbee-herdsman/tree/master/src/adapter/z-stack)

PS: FYI, see this other existing discussion for third-party tool for workaround to migrate between different types of radio adapters -> https://github.com/Koenkk/zigbee2mqtt/discussions/16478

**Comment by willieee802** [comment]
does it require repairing when switching from efr32 to 2652P adapter ?

**Reply by Hedda** [reply]
It depends, see https://github.com/Koenkk/zigbee2mqtt/discussions/16478

but also see https://github.com/Koenkk/zigbee2mqtt/discussions/23020

**Comment by lacrimal** [comment]
Shortly - is EFR32MG21 still considered experimental ?

**Reply by Hedda** [reply]
Not experimental if using the newer ”ember” driver (adapter) in Zigbee2MQTT and supported EmberZNet firmware which is currently version 7.4.x

https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

Note that the older ”ezsp” driver (adapter) that never made it out of experimental stage is now obsolete and deprecated.

https://www.zigbee2mqtt.io/guide/adapters/

Also note that newer EmberZNet 8.x.x firmware is still experimental (and older than 7.3.x not supported at all with the new ”ember” driver).

**Reply by lacrimal** [reply]
That's quite nice, since noticed info about ezsp deprecation - switched to new driver :-)

**Reply by Hedda** [reply]
Oh, and Multi-protocol / Multi-PAN RCP firmware is not currently supported either due to the various issues associated with it. So need to make sure you are using EmberZNet NCP firmware (and not RCP Multiprotocol firmware), which you probably are since EmberZNet NCP firmware has always been the standard, (unless you bought a SkyConnect dongle and been experimenting with Thread and multiprotocol).

**Comment by bobo-cher** [comment]
I have Sonoff Dongle-E with EFR32MG21 and use Ember driver with Z2M. All devices work well (48 devices). I have however one specific problem: OTA updates are usually fail to install. Currently 18 devices of the same type, Sonoff zbminir2) want to update to a newer firmware. During update (which normally takes 15 minutes) errors occur and the adapter is disconnected/reconnected followed by a restart of Z2M. After many attempts I've managed to update only few devices. I believe updates did work better when I had less devices in my network.

I wonder if a CC2652P based adapter (like Sonoff ZBDongle-P) will work any better when doing OTA updates?

**Reply by bobo-cher** [reply]
I tested many 7.x firmwares and all have the same issue (connection to coordinator getting lost during OTA updates). As a last resort tried old 6.10.3 firmware (ncp-uart-sw_EZNet6.10.3_V1.0.1.gbl) with ezsp driver. With this firmware OTA updates work without issues. Updated 6 devices without a single glitch.

Sonoff Dongle-E initially came with 6.x firmware (I'm not sure what version exactly it was), which I updated to 7.4 before use. Could it be that along with the update some other settings (variables etc.) need to be changed on the coordinator to work properly?

I would rather use a recent 7.x firmware but not working OTA updates is an issue.

**Reply by misery** [reply]
Did you tried the firmware from itead/sonoff, too?
https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP_7.4.3

**Reply by bobo-cher** [reply]
Yes, tried all itead firmwares (from repository you linked) and also 7.4.1-7.4.5 from https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee.

The only firmware where OTA updates work without issues is 6.10.3 from itead.

OTA updates did work with 7.4.x previously, when I had only few devices. But now, with 48 devices in the network, OTA updates almost always fail. They might work if I do a lot of attempts.

The issue isn't caused by the number of devices registered in the coordinator. I did a test with a second dongle, where only one device was registered and the issue is the same (all other devices were still actively spamming the air but the first coordinator was disconnected during tests). The second dongle was connected to a notebook (not to the main server). From that I can assume the problem isn't in the hardware (two dongles on two different computers). Both dongles can update successfully when using 6.10.3 firmware.

I have 11 devices which I haven't updated yet in a case developers want me to do further tests. I can always reproduce the issue.

**Reply by bobo-cher** [reply]
I have another observation.

In my previous tests with second dongle I was using parts of the z2m config file from my server. In particular network key and pan id/extended id were reused. I believe that makes the coordinator to accept/process packets sent by all devices in the house, even though they are not listed/visible in z2m on the second dongle/notebook.

I made another test, where I removed network key and pan id/extended id from the config of the second dongle. I guess that forces z2m to use some default or random values. In this test the OTA update (of the only one device in the new network) worked fine.

Thoughts after this test: I think 7.x firmware has difficulties processing all traffic in my network during OTA updates. I don't think it's a problem with one particular device. Difficulties with OTA updates sequentially increased with the grow of my network until they reached a point, where OTA updates become impossible at all.

BTW tried 8.0.2 firmware and got the same issue.
